### DISCUSSION TITLE
[Guide] How to join any hosted server in the most painless way (not for hosting and setting up a server)

### DISCUSSION THREAD

**Discussion by minhngoc25a** [Melledy/Grasscutter#589] [General]
# Grassclipper X

**Disclaimer: This guide assumes that you have correctly followed the [previous guide about setting up a server](https://github.com/Grasscutters/Grasscutter/discussions/713). You should worry about this if you're hosting the server with Grasscutter. You don't have to worry about this if your friend is the one inviting you to their server for multiplayer play or if you're joining a publicly-hosted server managed by someone else.**

Step 1: [Download](https://github.com/Grasscutters/GrassClipper-X) and extract the latest release of Grassclipper X.
Step 2: Launch Grassclipper X in the extracted folder.

![image](https://user-images.githubusercontent.com/33834488/167148729-d27cd87e-e2fa-4c45-b294-c1fa419f522d.png)

Step 3: Click on the Settings icon to the left of the minimize icon, choose the game executable (not the launcher, usually is under `./GenshinImpactGame` in the game root folder), and enter the server IP and the port.

![image](https://user-images.githubusercontent.com/33834488/167148886-18c6aec2-3189-46b7-8ab7-434e0c56b5c3.png)
*If you're hosting the server and playing on the same device or on the same network (this is called local play for short), the default IP should be localhost (or 127.0.0.1) with port 443. You can change this by editing config.json in your server root directory, though if you leave everything as it is, it should work as always.*

Step 4: Tick "Custom server," hit "Play," and enjoy!

![image](https://user-images.githubusercontent.com/33834488/167149030-d3d3ab0c-4552-44a6-a334-54e31778ab55.png)

---------------------

# Frequently asked questions

Q: Error 502??
A: If you have initially run some scripts from the original Grassclipper launcher, they might have installed a proxy for you. Turn off the manual proxy by going to the Settings app >> Network & Internet >> Proxy >> Turn off "Use a proxy server." Furthermore, there might be issues with the firewall, so turn it off or add exclusions for outbound and inbound blocking. Also, this:

![image](https://user-images.githubusercontent.com/33834488/167150284-3bce69c4-394f-4b07-a799-bd5302e02394.png)

# Other guides
[How to set up a server](https://github.com/Grasscutters/Grasscutter/discussions/713)

**Comment by GreatXuxu** [comment]
Yay! Thanks for your sharing.

**Reply by leafanine** [reply]
how do i get the public ip for the server

**Comment by AltoidDev** [comment]
when trying to join my server I get error 4214: "Unable to connect to server, please check your network settings."
